addons/knowledge/static/src/js/knowledge_controller.js

```diff
@@ -263,12 +275,40 @@ const KnowledgeFormController = FormController.extend({
         }
     },
 
+    /**
+     * @param {integer} id - Target id
+     * @returns {Promise}
+     */
+    _lock: async function (id) {
+        const { data } = this.model.get(this.handle);
+        const result = await this._rpc({
+            route: `/knowledge/article/${id}/${data.is_locked ? 'unlock' : 'lock'}`
+        });
+        if (result) {
+            if (!data.is_locked) {
+                await this._setMode('readonly');
+            } else {
+                await this.reload();
+            }
+        }
+    },
+
     // Helpers:
 
     /**
      * @override
      */
     _setMode: function () {
```

Comment:
even if the param is not used directly in the override, better to add it. So we know that we call it using param.